### PR TITLE
Added in comprehensive README and minor changes

### PR DESCRIPTION
**Pull request by rayryeng**
- Added in a comprehensive README that details what the project is about, how to use it, its purpose as well as training tips for training neural networks
- Fixed some typos in the docstring of the NeuralNet2 class
- Added some more comments in the demo code for the NeuralNet2 class, as well as ensuring that the path to the NeuralNet2 class is added
- Fixed `randperm` calls to allow prior versions of MATLAB to run the code
- Code has been verified and tested in versions from R2009b or later.

### CODE REVIEW

**Comment by rayryeng** [comment]
@amroamroamro This PR is not important... I just wanted to see that this project is more or less complete with a comprehensive README. I also fixed up some typos on my end and some fixes to allow more legacy versions of MATLAB to run our code.

**Comment by rayryeng** [comment]
@amroamroamro Gonna go ahead and merge this. I think this is OK. Hope you don't mind. Please update your version of the repo when you have the chance.

**Comment by amroamroamro** [comment]
@rayryeng  not at all, thanks for adding such comprehensive documentation!

Perhaps we should move some of it from the readme file into separate user-guide docs?
I'm thinking going with the repo's wiki. I've already used it for showing the published version of the demo we have: 
https://github.com/StackOverflowMATLABchat/NeuralNetPlayground/wiki

Many projects use the wiki this way (see [here](https://github.com/showcases/projects-with-great-wikis) for some examples); So each major section goes into a separate page. What do you think?

**Comment by rayryeng** [comment]
@amroamroamro  You're very welcome :) BTW, I've added the project on MATLAB FEX for more exposure. Hope you don't mind! Yes the wiki is actually a great idea. I'd love to get some feedback on what to put and what to leave out. Whenever you have time of course! I know you're busy. Let me know when you're free.

**Comment by amroamroamro** [comment]
@rayryeng Good idea with FEX, thanks for doing it.

I think the "overview of the GUI" section and its subsections can be moved to the wiki. "Sample Run" and "Training Tips" as well. Right?

I went ahead and created the first [overview page](https://github.com/StackOverflowMATLABchat/NeuralNetPlayground/wiki/Overview-of-the-GUI). You can do the same for the others, or if you want I'll finish copying the rest..

Next we can fix the "home" page by adding a table of contents and also a navigation sidebar on the left.

---

**UPDATE**: done, I moved things over to the [wiki](https://github.com/StackOverflowMATLABchat/NeuralNetPlayground/wiki). I also created a sidebar and footer for easy navigation.

**Comment by amroamroamro** [comment]
@rayryeng By the way I didn't have time to finish this from last time, but I had started working on some major changes for the neural net class (a version 2 if you will). It's a rewrite adding many options to tweak:
- separate activation functions for hidden/output layers
- choosing the loss function
- more initialization methods
- more activation and regularization functions
- more control over train stopping criteria
- optional gradient calculation checking using numerical estimation
- options to use more advanced external optimizers to find network weights that minimize the cost function (like `fminunc` and `fmincg`), in addition to our existing implementation of gradient descent (full and stochastic)

It's not ready just yet, but I will be continuing work where I left off as soon as I can ;)

**Comment by rayryeng** [comment]
@amroamroamro  sorry for not getting back to you immediately. I've taken a look at the Wiki and it looks like you went ahead and did the rest of it anyway :) hehe. Thank you. I would have done it myself but you went ahead and I do appreciate that a lot.  Also love the sidebar and footer.

As for v2 of the class, that's great. I'm eager to see what modifications you have made from the simple implementation that I came up with in the past.  Choosing the loss function as well allowing different activation functions and regularization functions would be great.

The gradient checking is a FANTASTIC idea. I didn't consider putting that in to ensure that the loss function is being minimized as the epochs evolve.  External optimizers are always great too because it will allow the loss function to be minimized more quickly.

Whenever you have time and when it's ready, let me know :) You can assign the changes to me as  PR or you can merge onto master yourself. I trust you.

Thanks for continuing to improve this! I like where this is going and I'm glad we're finally collaborating together in a project.
